frappe/permissions.py

```diff
@@ -333,14 +333,14 @@ def check_user_permission_on_link_fields(d):
 				# restricted for this link field, and no matching values found
 				# make the right message and exit
 				if d.get("parentfield"):
-					# "Not allowed for Company = Restricted Company in Row 3. Restricted field: reference_type"
-					msg = _("Not allowed for {0}: {1} in Row {2}. Restricted field: {3}").format(
-						_(field.options), d.get(field.fieldname), d.idx, field.fieldname
+					# "You are not allowed to access Company 'Restricted Company' in row 3, field Reference Type"
+					msg = _("You are not allowed to access {0} '{1}' in row {2}, field {3}").format(
+						_(field.options), d.get(field.fieldname) or _("empty"), d.idx, _(field.label)
 					)
 				else:
-					# "Not allowed for Company = Restricted Company. Restricted field: reference_type"
-					msg = _("Not allowed for {0}: {1}. Restricted field: {2}").format(
-						_(field.options), d.get(field.fieldname), field.fieldname
+					# "You are not allowed to access Company 'Restricted Company' in field Reference Type"
+					msg = _("You are not allowed to access {0} '{1}' in field {2}").format(
+						_(field.options), d.get(field.fieldname) or _("empty"), _(field.label)
```

Comment:
we've started removing field labels with tab layout (i think) - might not be good in those cases as the label will turn out to be empty - maybe we can keep the field as a fallback?

(same for above)

Reply:
I'd be surprised if we have any restricted link fields without a label, but I added the field name as a fallback.